### ISSUE TITLE
Potentially unintended text in search on docs

### ISSUE THREAD

**Issue by morganarnel** [twbs/bootstrap#37312]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

Go to the docs page at: https://getbootstrap.com/docs/5.2/customize/overview/
Press `/` to open the Algolia search, when searching for `Modal` what are presumably supposed to be results are displayed as a heading.

![image](https://user-images.githubusercontent.com/84181964/195575194-e19ebf8b-51b9-4ef1-b77f-d45c3d55e5d4.png)

### Reduced test cases

NA

### What operating system(s) are you seeing the problem on?

Windows, macOS, Android, iOS, Linux

### What browser(s) are you seeing the problem on?

Chrome, Firefox, Microsoft Edge

### What version of Bootstrap are you using?

NA

**Comment by julien-deramond** [comment]
Apparently, all the `<h1>` [in the Modal page](https://getbootstrap.com/docs/5.2/components/modal/) are gathered and appear in the results as a title in violet.

**Comment by WinterSilence** [comment]
[customize widget](https://www.algolia.com/doc/api-reference/widgets/autocomplete/js/)

**Comment by mdo** [comment]
@patrickhlauke We may want to revisit the h1 move in modal headers.

**Comment by patrickhlauke** [comment]
so is there no way to get the silly algolia thing to exclude certain types of headings with a particular `.modal-title` class? would hate to have to compromise correct structure/markup just for this...

**Comment by julien-deramond** [comment]
I'm not confortable to compromise the correct structure. We could create an issue in their repo so they use only the first h1 of the page for this part 🤷

**Comment by WinterSilence** [comment]
in `search.js`:
~~~js
...
transformItems(items) {
  return items.map(item => {
    item._highlightResult.hierarchy.lvl0.value = '' // Added
    const liveUrl = 'https://getbootstrap.com/'
    ...
  })
}
...
~~~

**Comment by julien-deramond** [comment]
Hum. If we do it, we'll lose some information in the results (violet titles).

Before:

<img width="574" alt="Screenshot 2022-10-17 at 11 24 17" src="https://user-images.githubusercontent.com/17381666/196141306-07a494f3-4a41-4957-bcec-8fd52fc62fe2.png">

With the proposed patch:

<img width="574" alt="Screenshot 2022-10-17 at 11 24 28" src="https://user-images.githubusercontent.com/17381666/196141408-87fc27b9-75da-4708-9a50-35088d8ebaf5.png">

**Comment by WinterSilence** [comment]
@julien-deramond it's rude solution just for example

I explode default configs and find solution in [config.js](https://docsearch.algolia.com/docs/templates#default-template):
~~~js
  ...
  actions: [
    {
      recordExtractor: ({ helpers }) => {
        return helpers.docsearch({
          recordProps: {
            lvl1: ['header h1', 'article h1', 'main h1', 'h1', 'head > title'],
            content: ['article p, article li', 'main p, main li', 'p, li'],
            lvl0: {
              selectors: '',
              defaultValue: 'Documentation',
            },
            lvl2: ['article h2', 'main h2', 'h2'],
            lvl3: ['article h3', 'main h3', 'h3'],
            lvl4: ['article h4', 'main h4', 'h4'],
            lvl5: ['article h5', 'main h5', 'h5'],
            lvl6: ['article h6', 'main h6', 'h6'],
          },
        });
      },
    },
  ],
  ...
~~~

**Comment by julien-deramond** [comment]
Yes it could be the solution [with this syntax](https://docsearch.algolia.com/docs/record-extractor/#manipulate-the-dom-with-cheerio):
```js
recordExtractor: ({ $, helpers }) => {
  // Removing DOM elements we don't want to crawl
  $(".modal-title").remove();

  return helpers.docsearch({
    // ...
  });
},
```

**Comment by bastienmoulia** [comment]
Same with `ti` (would like to find titles) :
![image](https://user-images.githubusercontent.com/686196/216353491-635c62f4-0a7d-4083-9b5d-71fbede1e3bb.png)

**Comment by coliff** [comment]
Maybe this issue can be closed now as the Search for v6 is powered by [Pagefind](https://pagefind.app/) (and this issue is fixed there)?

**Comment by julien-deramond** [comment]
Yep, let's close as it's v5-specific, and we won't probably have time to fix it.
